jaxlib/cuda/versions_helpers.cc

```diff
@@ -85,4 +85,11 @@ size_t CudnnGetVersion() {
   return version;
 }
 
+int PtxasGetVersion(){
+  unsigned int major, minor;
+  JAX_THROW_IF_ERROR(JAX_AS_STATUS(nvPTXCompilerGetVersion(&major, &minor)));
```

Comment:
This is actually checking the version of the `nvptxcompiler` library, which is a static library containing the ptx compiler. But we don't use that! We use a `ptxas` *binary*. And nothing requires that the versions of each would be the same. So any version check would need to work by invoking `ptxas`, as XLA itself does.

One way we could handle this would be to have *JAX* decide on which `ptxas` to use, not XLA, and to check the `ptxas` version. Further: there's no particular reason that logic is in C++: we're just shelling out to a subprocess. So we could probably make it 20 lines of relatively readable Python, rather than having to touch this code.

We could then simply pass that path to XLA explicitly, rather than having XLA search for it. That would be simpler and perhaps more readable.


(Now, it's also possible that we should stop using a `ptxas` binary, instead linking in `libnvptxcompiler` into XLA. That might be possible, and would solve the problem another way. The main downside of this is I'd want to verify that we don't, e.g., break things like concurrent compilation. I'd be tempted to go with the first option for now.)